src/python/turicreate/data_structures/sframe.py

```diff
@@ -2385,11 +2385,18 @@ def to_dataframe(self):
         out : pandas.DataFrame
             The dataframe which contains all rows of SFrame
         """
+
+        from ..toolkits.image_classifier._evaluation import _image_resize
+
         assert HAS_PANDAS, 'pandas is not installed.'
         df = pandas.DataFrame()
         for i in range(self.num_columns()):
             column_name = self.column_names()[i]
-            df[column_name] = list(self[column_name])
+            if(self.column_types()[i] == _Image and explore):
+                pandas.set_option('display.max_colwidth', -1)
+                df[column_name] = [_image_resize(x[column_name])._to_pil_image() for x in self.select_columns([column_name])]
+            else:
+                df[column_name] = list(self[column_name])
```

Comment:
Is there not a proper way to put a PIL image inside a Pandas dataframe? That could allow us to reconcile these two codepaths and just have the dataframe actually contain the image.

Reply:
Check out https://www.kaggle.com/stassl/displaying-inline-images-in-pandas-dataframe for an example of storing the image directly in the dataframe.